tools/code_samples/code_samples_usage.php

```diff
@@ -0,0 +1,213 @@
+<?php
+
+if (1 < $argc) {
+    $codeSampleFileList = array_slice($argv, 1);
+    foreach ($codeSampleFileList as $codeSampleFile) {
+        echo "\n$codeSampleFile\n\n";
+        $includingFileList = getIncludingFileList($codeSampleFile);
+        foreach ($includingFileList as $includingFile) {
+            $blocks = getInclusionBlocks($includingFile, $codeSampleFile);
+            displayBlocks($blocks, $includingFile);
+        }
+    }
+} else {
+    $includingFileList = getIncludingFileList();
+    foreach ($includingFileList as $includingFile) {
+        $blocks = getInclusionBlocks($includingFile);
+        displayBlocks($blocks, $includingFile);
+    }
+}
+
+function displayBlocks(array $docFileBlocks, string $docFilePath = null, $lineOffset = 0): void
+{
+    if (!$docFilePath) {
+        $docFilePath = '';
+    }
+    foreach ($docFileBlocks as $block) {
+        $prefixedBlockLines = [];
+        foreach ($block as $blockLineIndex => $blockLine) {
+            $lineNumber = $blockLineIndex + $lineOffset;
+            $prefixedBlockLines[$blockLineIndex] = "$docFilePath@$lineNumber:$blockLine";
+        }
+        echo implode(PHP_EOL, $prefixedBlockLines) . PHP_EOL . PHP_EOL;
+
+        $prefixedBlockContentLines = [];
+        try {
+            $blockContents = getBlockContents($block);
+            foreach ($blockContents['contents'] as $contentLineNumber => $contentLine) {
+                $prefixedBlockContentLines[] = str_pad($contentLineNumber, 3, 0, STR_PAD_LEFT) . (in_array($contentLineNumber, $blockContents['highlights']) ? '⫸' : '⫶') . $contentLine;
+            }
+            echo implode(PHP_EOL, $prefixedBlockContentLines) . PHP_EOL . PHP_EOL;
+        } catch (Exception $exception) {
+            echo $exception->getMessage() . PHP_EOL . PHP_EOL;
+        }
+    }
+}
+
+
+// FRAMEWORK
+
+/**
+ * @param string $targetBranch Branch to compare to.
+ * @param string $sourceBranch Branch to compare from. By default, the current branch HEAD
+ * @return array<int, string> List of file paths from code_samples/ directory.
+ */
+function getModifiedCodeSampleFileList(string $targetBranch = 'origin/master', string $sourceBranch = 'HEAD'): array
```

Comment:
Is this function used? I don't see the usage

Reply:
You're right. It became unused somewhen in the writing, I had to move to Github PR API. I'll clean this up.

Reply:
Removed in d09fd94d081b88183f01f8ff29cf14cb7554acb4